CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/DefaultInteractionEventHandler.java

```diff
@@ -1006,21 +1006,13 @@ private void performDrag(GraphWriteMethods wg, final Camera camera, final Point
 
         final Vector3f delta = visualInteraction.convertTranslationToDrag(camera, position, from, to);
 
-        final int xAttribute = VisualConcept.VertexAttribute.X.get(wg);
-        final int yAttribute = VisualConcept.VertexAttribute.Y.get(wg);
-        final int zAttribute = VisualConcept.VertexAttribute.Z.get(wg);
-        final int x2Attribute = VisualConcept.VertexAttribute.X2.get(wg);
-        final int y2Attribute = VisualConcept.VertexAttribute.Y2.get(wg);
-        final int z2Attribute = VisualConcept.VertexAttribute.Z2.get(wg);
-        final int cameraAttribute = VisualConcept.GraphAttribute.CAMERA.get(wg);
-        
         draggedNodeIds.forEach(vertexId -> {
-            final Vector3f currentPos = VisualGraphUtilities.getMixedVertexCoordinates(wg, vertexId, xAttribute, x2Attribute, yAttribute, y2Attribute, zAttribute, z2Attribute, cameraAttribute);
+            final Vector3f currentPos = VisualGraphUtilities.getMixedVertexCoordinates(wg, vertexId);
             currentPos.add(delta);
-            VisualGraphUtilities.setVertexCoordinates(wg, currentPos, vertexId, xAttribute, yAttribute, zAttribute);
+            VisualGraphUtilities.setVertexCoordinates(wg, currentPos, vertexId);
         });
 
-        scheduleXYZChangeOperation(Ints.toArray(draggedNodeIds));
```

Comment:
I think here that these draggedNodeIds should be positions and not Id's. I reckon this only becomes an issue when some vertices are deleted. This would explain why there is an outOfBounds error occuring and how it slipped through previous testing and yet seems fairly widespread?